src/app/dashboard/dashboard.component.spec.ts

```diff
@@ -0,0 +1,183 @@
+
+describe('Dashboard', () => {
+  @Component({
+    template: '<tk-dashboard></tk-dashboard>'
+  })
+  class TestComponent {}
+
+  let fixture: ComponentFixture<TestComponent>;
+  let component: DashboardComponent;
+  let harness: DashboardHarness;
+  let store: MockStore<AppState>;
+
+  const mockProductService = {
+    getProducts: jasmine
+      .createSpy('getProducts')
+      .and.returnValue(of(mockProductResponse))
+  };
+
+  const mockChartProductData = {
+    fieldNames: ['Task', 'Hours per Day'],
+    data: [
+      ['Work', 11],
+      ['Eat', 2],
+      ['Commute', 2],
+      ['Watch TV', 2],
+      ['Sleep', 7]
+    ]
+  };
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [NoopAnimationsModule, RouterTestingModule],
+      declarations: [
+        TestComponent,
+        DashboardComponent,
+        MockPipe(PieChartProductDataPipe, () => mockChartProductData),
+        MockPipe(BarChartProductDataPipe, () => mockChartProductData),
+        MockComponent(ChartComponent)
+      ],
+      providers: [
+        provideMockStore({ initialState }),
+        MockProvider(ProductService, mockProductService)
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+
+    store = TestBed.inject(MockStore);
+    spyOn(store, 'dispatch');
+
+    fixture = TestBed.createComponent(TestComponent);
+    component = fixture.debugElement.query(By.directive(DashboardComponent))
+      .componentInstance;
+
+    const loader = TestbedHarnessEnvironment.loader(fixture);
+    harness = await loader.getHarness(DashboardHarness);
+
+    fixture.detectChanges();
+  });
+
+  it('should be created', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should display loader when loading is true', async () => {
+    expect(await harness.isLoading()).toBeFalse();
+    component.loading$.next(true);
+    expect(await harness.isLoading()).toBeTrue();
+  });
+
+  it('should display page title', async () => {
+    const receivedTitle = await harness.pageTitle();
+    const expectedTitle = 'Product Info Dashboard';
+
+    expect(receivedTitle).toBe(expectedTitle);
+  });
+
+  it('should display text when product not selected', async () => {
+    const pageText = await harness.pageText();
+    expect(pageText).not.toBeNull();
+    expect(
+      fixture.debugElement.queryAll(By.directive(ChartComponent)).length
+    ).toBe(0);
+  });
+
+  it('should dispatch products retrieval on component creation', () => {
+    expect(store.dispatch).toHaveBeenCalledWith(
+      ProductActions.retrieveProducts(mockProductResponse)
+    );
+  });
+
+  describe('Pie Chart', () => {
+    const targetProduct = initialState.product.products[0];
+    let pieChartComponent: ChartComponent;
+
+    beforeEach(() => {
+      store.overrideSelector(ProductSelectors.search, targetProduct.name);
+      store.refreshState();
+      fixture.detectChanges();
+
+      pieChartComponent = fixture.debugElement.queryAll(
```

Comment:
define harness for chartComponent and find pie/bar charts suing it.

Reply:
I don't see a way to use a harness for this particular set of tests. The chart component is mocked and all I test is if it's rendered and if props are passed correctly. All I can do with a harness here is to check if the component is present.
Or what you suggest is to test only the presence and dump the rest of the suite as unnecessary?

Reply:
Sorry I haven't notice that you actually mocked ChartComponent. 
We are not mocking child components. in that way we can test actual component behaviour. 
But based on the way you implemented chart component, I agree, the way you locate and test it the only best possible way.